### ISSUE TITLE
Self Hosted Videos

### ISSUE THREAD

**Issue by RambosRide** [nanostudio-org/nanogallery2#147]
I host my own short 10-15 second videos I'd like to use in gallery view.

Don't want to have to host externally.

Is this a feature you'd be willing to implement? it really is just changing from iframe output to an html video player?

Thanks

Dan

**Comment by Kris-B** [comment]
this is an open source project, are you interested to implement it ?

**Comment by willus** [comment]
I second this.  I'd very much like to do self-hosted .mp4 videos with nanogallery2.  Shouldn't be too hard--all major browsers support .mp4 natively now.  I'd be willing to try and hack around in the js source, but I don't even know where to start.  I just need a hack to have the video auto-play in display mode.  Doesn't have to be fancy.

**Comment by interwap** [comment]
**Self Hosted Video Support:**

Hi Guys, so i played with jquery.nanogallery2.js and added the selfhosted video  functionality. This checks the file extension and renders an html5 video tag instead of an iframe.
Here is the link to the Gist (patched file): [https://gist.github.com/interwap/66f7c19281a579458fbee0576cce6ea9](url)

// Changes made:
Line 5751 - 5778 (Added self-hosted video property to medialist)
Line 5910  - 5916 (implemented selfhosted video check via url extension)
Line 6126  - 6132 (Repeated the above)

I hope the main contributors of the library can look at this implementation and add it core.
Hope you guys find this somewhat useful...

**Comment by Kris-B** [comment]
@interwap thanks for the implementation ;-)

**Comment by Kris-B** [comment]
self hosted videos (mp4) are supported in v2.4.0 - https://github.com/nanostudio-org/nanogallery2/releases/tag/v2.4.0

**Comment by sellleon** [comment]
In the test I set up there was issue with playing the video that I clicked on. There seems to be an invisible layer above the video preventing me from clicking on play. Though if click on the arrow button to switch to the next video that layer seems to be gone and I can click on play just fine. Tested on both firefox and chromium.

**Comment by Kris-B** [comment]
@sellleon can you share your page where this issue can be reproduced ?

**Comment by Julisoft2019** [comment]
Any example of a video file code on the host? ... (v2.4.1) 

"itemsBaseURL": "http://www.website.cl/videos/",
<a href="video.mp4" data-ngthumb="video_thumb.jpg" data-ngdesc="">Video 01</a>

It does not work for me !!!
Thank you.

**Comment by Kris-B** [comment]
```
<div id="my_nanogallery2" data-nanogallery2 = '{ "itemsBaseURL": "https://mywebserver/gallery/content/"}' >
  <a href="my_video.mp4" data-ngthumb="my_video_thumbnail.jpg" >title video</a>
  <a href="img_01.jpg" data-ngthumb="img_01t.jpg">Title Image1</a>
</div>
```

**Comment by Julisoft2019** [comment]
Hi,
It does not work with video ... only with image works well.    :(

**Comment by sellleon** [comment]
@Kris-B Here's a demo which has the issue I described. Library js and css files are in the correct place. Media files as well. Playback works correctly but only after clicking on an arrow button first as described.
`demo.html`:
```
<!DOCTYPE HTML>
<html>
<head>
    <link rel='stylesheet' href='css/nanogallery2.woff.min.css'>
</head>

<body>
<div id="my_nanogallery2_video"></div>

<script src='js/jquery.min.js'></script>
<script src='js/jquery.nanogallery2.min.js'></script>
<script src="js/demo.js"></script>
</body>
</html>
```
`js/demo.js`:
```
jQuery(document).ready(function () {
  jQuery("#my_nanogallery2_video").nanogallery2({
  	items:[
      {
        src:          'v1.mp4',
        srct:         'v1.jpg',
        title:        'Video',
      },
      {
        src:          'v2.mp4',
        srct:         'v2.jpg',
        title:        'Video',
      },
      {
        src:          'v3.mp4',
        srct:         'v3.jpg',
        title:        'Video',
      },
      {
        src:          'v4.mp4',
        srct:         'v4.jpg',
        title:        'Video',
      }
		],
        itemsBaseURL: "http://www.example.org/",
        thumbnailWidth: "auto",
        thumbnailBorderVertical: 2,
        thumbnailBorderHorizontal: 2,
        thumbnailLabel: {
          display: false
        },
        colorScheme: {
          thumbnail: {
            borderColor: "rgba(255,255,255,1)"
          }
        },
        thumbnailAlignment: "center"
  });
});
```

**Comment by Julisoft2019** [comment]
HTML markup example?

**Comment by Kris-B** [comment]
@sellleon fixed in v2.4.2

**Comment by Kris-B** [comment]
@Julisoft2019 html markup issue fixed in v2.4.2

```
<div id="my_nanogallery2" data-nanogallery2 = '{ "itemsBaseURL": "https://mywebserver/gallery/content/"}' >
  <a href="my_video.mp4" data-ngthumb="my_video_thumbnail.jpg" >title video</a>
  <a href="img_01.jpg" data-ngthumb="img_01t.jpg">Title Image1</a>
</div>
```
